nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/QueryRowsMarkLogic.java

```diff
@@ -24,85 +24,106 @@
+	public void onTrigger(ProcessContext context, ProcessSessionFactory sessionFactory) throws ProcessException {
+		final ProcessSession session = sessionFactory.createSession();
+		FlowFile flowFile = session.get();
+		FlowFile incomingFlowFile = flowFile;
+		try {
+			
+			if (flowFile == null) {
+				flowFile = session.create();
+			}
+
+			final String jsonPlan = determineJsonPlan(context, flowFile);
+			final String mimeType = determineMimeType(context, flowFile);
+
+			session.putAttribute(flowFile, "marklogic-optic-plan", determineJsonPlan(context, flowFile));
+
+			final DatabaseClient client = getDatabaseClient(context);
+			final RowManager rowManager = client.newRowManager();
+			PlanBuilder.Plan plan = rowManager.newRawPlanDefinition(new StringHandle(jsonPlan));
+
+			InputStream inputStream = rowManager.resultDoc(plan, new InputStreamHandle().withMimetype(mimeType)).get();
+			if (inputStream != null) {
+				try {
+					if(incomingFlowFile != null) {
+						incomingFlowFile = session.write(flowFile, out -> {
+							FileCopyUtils.copy(inputStream, out);
+						});
+					} else {
+						flowFile = session.write(flowFile, out -> {
+							FileCopyUtils.copy(inputStream, out);
+						});
+					}
+				} finally {
+					inputStream.close();
+				}
+			}
+			if(incomingFlowFile != null) {
+				transferAndCommit(session, incomingFlowFile, SUCCESS);
+			}
+
+			if (flowFile != null) {
+                transferAndCommit(session, flowFile, ORIGINAL);
             }
-
-            final String jsonPlan = determineJsonPlan(context, flowFile);
-            final String mimeType = determineMimeType(context, flowFile);
-
-            final DatabaseClient client = getDatabaseClient(context);
-            final RowManager rowManager = client.newRowManager();
-            PlanBuilder.Plan plan = rowManager.newRawPlanDefinition(new StringHandle(jsonPlan));
-
-            InputStream inputStream = rowManager.resultDoc(plan, new InputStreamHandle().withMimetype(mimeType)).get();
-            if (inputStream != null) {
-                try {
-                    flowFile = session.write(flowFile, out -> {
-                        FileCopyUtils.copy(inputStream, out);
-                    });
-                } finally {
-                    inputStream.close();
-                }
-            }
-
-            transferAndCommit(session, flowFile, SUCCESS);
-        } catch (final Throwable t) {
-            this.logErrorAndRollbackSession(t, session);
-        }
-    }
-
-    protected String determineJsonPlan(ProcessContext context, FlowFile flowFile) {
-        return context.getProperty(PLAN).evaluateAttributeExpressions(flowFile).getValue();
-    }
-
-    protected String determineMimeType(ProcessContext context, FlowFile flowFile) {
-        return context.getProperty(MIMETYPE).evaluateAttributeExpressions(flowFile).getValue();
-    }
+			
+		} catch (final Throwable t) {
+			// getLogger().error("markLogicErrorMessage", new Object[] { this }, t);
```

Comment:
We generally don't want to leave commented-out lines in place - just remove this line.